graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -22,14 +31,15 @@
     ),
 )
 
-# IntegerInterval is used to denote a continuous non-empty interval of integers.
-IntegerInterval = namedtuple(
-    "IntegerInterval",
+# Interval is used to denote a continuous non-empty interval of values.
```

Comment:
I could make it generic, but then it wouldn't be a namedtuple. Mypy doesn't support generic namedtuples.

Reply:
Doing this in a separate PR.. Looks like it's worth it

Reply:
https://github.com/kensho-technologies/graphql-compiler/pull/726

Reply:
Generic dataclasses are a thing, though — and I think that might be worth looking into.